rest_api/rest_api/pipeline/__init__.py

```diff
@@ -24,39 +25,44 @@
 UNSUPPORTED_DOC_STORES = (FAISSDocumentStore, InMemoryDocumentStore)
 
 
-def setup_pipelines() -> Dict[str, Any]:
+def setup_pipelines(pipeline_hyper_params: PipelineHyperParams) -> Dict[str, Any]:
     # Re-import the configuration variables
     from rest_api import config  # pylint: disable=reimported
 
     pipelines = {}
 
-    # Load query pipeline
-    # query_pipeline = Pipeline.load_from_yaml(Path(config.PIPELINE_YAML_PATH), pipeline_name=config.QUERY_PIPELINE_NAME)
-
     # ------------------
-    document_store = ElasticsearchDocumentStore(host="host.docker.internal")
+    document_store = ElasticsearchDocumentStore()
     extractive_document_store = ElasticsearchDocumentStore(
-        host="host.docker.internal", index="rulebook", embedding_dim=768
+        index="rulebook",
+        embedding_dim=pipeline_hyper_params.extractive_embedding_dim,
+        similarity=pipeline_hyper_params.extractive_similarity_function,
     )
     faq_document_store = ElasticsearchDocumentStore(
-        host="host.docker.internal", index="faq", embedding_dim=384, similarity="cosine"
+        index="faq",
+        embedding_dim=pipeline_hyper_params.faq_embedding_dim,
+        similarity=pipeline_hyper_params.faq_similarity_function,
     )
 
-    extractive_reader_option = "deepset/roberta-base-squad2"
-    faq_retriever_option = "sentence-transformers/all-MiniLM-L6-v2"
+    extractive_reader_option = pipeline_hyper_params.extractive_reader_option
```

Comment:
I think both of these can be directly replaced by `pipeline_hyper_params` in their retriever's definition

Reply:
https://github.com/rafaljanwojcik/board-game-rules-explainer/pull/4/commits/7b1b47982f8b446bbb125918a2b3cc8b9212f5cd